server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1050,4 +1006,91 @@ private void cleanUpCompactors() {
+      LOG.debug("Removed stale entries from RUNNING_CACHE : {}", idsToRemove);
+    }
+
+    // Get the set of groups being referenced in the current configuration
+    // Needs Dan's changes for this
+    Set<CompactorGroupId> groupsInConfiguration = new HashSet<>();
+
+    // Compaction jobs are created in the TabletGroupWatcher and added to the Coordinator
+    // via the addJobs method which adds the job to the CompactionJobQueues object.
+    Set<CompactorGroupId> groupsWithJobs = jobQueues.getQueueIds();
+
+    Set<CompactorGroupId> jobGroupsNotInConfiguration =
+        Sets.difference(groupsWithJobs, groupsInConfiguration);
+
+    if (jobGroupsNotInConfiguration != null && !jobGroupsNotInConfiguration.isEmpty()) {
+      RUNNING_CACHE.values().forEach(rc -> {
+        if (jobGroupsNotInConfiguration.contains(CompactorGroupId.of(rc.getGroupName()))) {
+          LOG.warn(
+              "External compaction {} running in group {} on compactor {},"
+                  + " but group not found in current configuration. Failing compaction...",
+              rc.getJob().getExternalCompactionId(), rc.getGroupName(), rc.getCompactorAddress());
+          cancelCompactionOnCompactor(rc.getCompactorAddress(),
+              rc.getJob().getExternalCompactionId());
+        }
+      });
+
+      // Remove groups not in configuration from TIME_COMPACTOR_LAST_CHECKED
+      LOG.debug("No longer tracking compactor check-in times for groups: {}",
+          jobGroupsNotInConfiguration);
+      jobGroupsNotInConfiguration.forEach(TIME_COMPACTOR_LAST_CHECKED::remove);
+    }
+
+    final Set<CompactorGroupId> runningCompactorGroups = new HashSet<>();
+    getRunningCompactors().keySet()
+        .forEach(group -> runningCompactorGroups.add(CompactorGroupId.of(group)));
+
+    Set<CompactorGroupId> groupsWithNoCompactors =
+        Sets.difference(groupsInConfiguration, runningCompactorGroups);
+    if (groupsWithNoCompactors != null && !groupsWithNoCompactors.isEmpty()) {
+      LOG.warn("The following groups have no running compactors: {}", groupsWithNoCompactors);
+    }
+
+    Set<CompactorGroupId> compactorsWithNoGroups =
+        Sets.difference(runningCompactorGroups, groupsInConfiguration);
+    if (compactorsWithNoGroups != null && !compactorsWithNoGroups.isEmpty()) {
+      LOG.warn(
+          "The following groups have running compactors, but are not in the current configuration: {}",
+          compactorsWithNoGroups);
+    }
+
+    long now = System.currentTimeMillis();
+    Map<String,Set<HostAndPort>> idleCompactors = getIdleCompactors();
+    TIME_COMPACTOR_LAST_CHECKED.forEach((groupName, lastCheckTime) -> {
+      if ((now - lastCheckTime) > getMissingCompactorWarningTime()
+          && jobQueues.getQueuedJobs(groupName) > 0
+          && idleCompactors.containsKey(groupName.canonical())) {
+        LOG.warn("No compactors have checked in with coordinator for group {} in {}ms", groupName,
+            getMissingCompactorWarningTime());
```

Comment:
This is detecting more than the absence of contact from compactors, its detecting that compactors seem to exists but are not doing work or checking in.  Tried to adjust the log message to reflect this.

```suggestion
        LOG.warn("The group {} has queued jobs and {} idle compactors, however none have checked in with coordinator for {}ms", groupName, idleCompactors.get(groupName).size()
            getMissingCompactorWarningTime());
```

Reply:
Changed log message in 32ad33e